#Taller Semana 11

Considerando la base de datos adjunta al taller debes completar los siguientes pasos, recuerda responder agregando si es necesario un cuadro de código y/o texto.

1. Anota las librerias necesarias para correr el(los) código(s).
2. Importa y limpia (si amerita) adecuadamente la base de datos.
3. Desarrolla un modelo de regresión lineal múltiple donde la variable explicada sea el producto interno bruto del país (gdp, por sus siglas en inglés).
4. ¿Cuál es el efecto causal que quieres estimar? ¿Por qué es interesante?
5. Concluya de manera global sobre los resultados del modelo.
6. Concluya de manera particular los resultados del módelo.
7. ¿Qué variables no consideradas podrían afectar la variable explicada? Justifique.

1. Anota las librerías necesarias para correr el(los) código(s).

In [ ]:
#libreria base para el código
import pandas as pd
#análisis
import statsmodels.api as sm
import statsmodels.formula.api as smf

2 . Importa y limpia (si amerita) adecuadamente la base de datos.

In [ ]:
#agregar y limpiar base
bdd = pd.read_csv("nations_gdp_school_life_co2.csv", sep = ';', decimal=",", index_col=0)
#para limpiar la base de datos quitaremos todos los paises que tengan NaN en alguna de sus columnas, y quitaremos la columna paises
df1 = bdd[["region", 'gdp', 'school', 'life', 'co2']].dropna()
#mostramos la estructura de la bdd, y la bdd filtrada
print(df1.info())
print(df1)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 175 entries, 1 to 194
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   region  175 non-null    object 
 1   gdp     175 non-null    float64
 2   school  175 non-null    float64
 3   life    175 non-null    float64
 4   co2     175 non-null    float64
dtypes: float64(4), object(1)
memory usage: 8.2+ KB
None
      region           gdp     school       life    co2
1     Africa   7300.399902   6.716667  72.316666  15.00
2     Africa   1338.800049   3.100000  54.733334   1.20
3     Africa  12307.400390   8.600000  52.250000   9.20
4     Africa   1063.400024   1.300000  53.783333   0.20
5     Africa    349.200012   2.483333  48.866665   0.10
..       ...           ...        ...        ...    ...
189  Oceania   1953.800049   4.116667  61.483334   2.75
190  Oceania   4012.600098  10.300000  71.533333   3.10
191  Oceania   2249.199951   4.500000  66.500000   1.40
192  Oceania   4072.

In [ ]:
#Anexo, estudio descriptivo de la variable
df1["gdp"].describe()

count      175.000000
mean     12262.840660
std      14056.937687
min        279.799988
25%       2100.300049
50%       6932.799805
75%      17804.200195
max      74906.000000
Name: gdp, dtype: float64

3. Desarrolla un modelo de regresión lineal múltiple donde la variable explicada sea el producto interno bruto del país (gdp, por sus siglas en inglés)

In [ ]:
#definimos las varibles para el modelo, para este modelo no consideraremos las regiones
gdp = df1["gdp"] #variable principal, pib
school = df1["school"]  #indice de escolaridad
life = df1["life"]  #esperanza de vida
co2= df1["co2"]  #toneladas de co2 producidas
lm1 = smf.ols(formula="gdp~school+life+co2", data=df1).fit()
lm1.params
print(lm1.summary())

                            OLS Regression Results                            
Dep. Variable:                    gdp   R-squared:                       0.769
Model:                            OLS   Adj. R-squared:                  0.764
Method:                 Least Squares   F-statistic:                     189.2
Date:                Tue, 27 Dec 2022   Prob (F-statistic):           4.26e-54
Time:                        21:09:09   Log-Likelihood:                -1791.2
No. Observations:                 175   AIC:                             3590.
Df Residuals:                     171   BIC:                             3603.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept  -1.983e+04   4250.157     -4.665      0.0

4. ¿Cuál es el efecto causal que quieres estimar? ¿Por qué es interesante?
---
<p align="justify">Se pretende analizar un modelo, que relacione el pib, con los años promedio de escolaridad, la esperanza de vida y las toneladas de CO2 por país.
Es una medida importante, porque refleja el nivel de actividad económica en una economía y, por tanto, puede ser utilizada para evaluar el bienestar de la población.
Además, el PIB también es útil para los gobiernos y los economistas a la hora de tomar decisiones económicas.


5. Concluya de manera global sobre los resultados del modelo.
---
<p align="justify">Al realizar el summary, notamos que el <b>R ajustado -Adj. R-squared-</b> es un 0.764, lo que quiere decir que las varaibles school, life y co2 <i>explican en un 76,4% los resultados de la variable gdp</i>, lo que nos indica que es un buen modelo<br>
Y como el <b>p estadistico -Prob(F-stastistic)-</b>, <i>no supera el 5%, el modelo es significativo</i>

6. Concluya de manera particular los resultados del módelo.
---
<p align="justify">En lo que respecta al analisis de las variables, el indice de escolaridad, la esperanza de vida y las toneladas de CO2, <b>son significativas</b> al no ser mayores que el 5%.

7.	¿Qué variables no consideradas podrían afectar la variable explicada? Justifique.
---
<p align="justify">Hay varias variables no consideradas, como la población total activa, ya que incide dentro del consumo privado, otra es el indice de libertad económica ya que incide en la cantidad de exportaciones e importaciones, o el indice de democracia, que tambien incide en el gasto público, todas varibles necesarias para la determinacion del PIB



Enlace al código:<br>
https://colab.research.google.com/drive/1N4ikygMDf6jnK7wFD39mblZ9OPctd8Hd?usp=sharing